lldb/source/Target/PathMappingList.cpp

```diff
@@ -264,34 +288,35 @@ PathMappingList::FindFile(const FileSpec &orig_spec) const {
 
 bool PathMappingList::Replace(llvm::StringRef path, llvm::StringRef new_path,
                               bool notify) {
-  std::lock_guard<std::recursive_mutex> lock(m_mutex);
-  uint32_t idx = FindIndexForPath(path);
-  if (idx < m_pairs.size()) {
+  {
+    std::lock_guard<std::mutex> lock(m_pairs_mutex);
+    uint32_t idx = FindIndexForPath(path);
```

Comment:
@JDevlieghere  It seems these two lines lead to deadlock. `FindIndexForPath` attempts to lock `m_pairs_mutex` too, but can't.

Reply:
Yup, I fixed the issue in the re-land. @kastiglione confirmed offline that that fixed the deadlock he observed.